x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/store/RoleDescriptorStore.java

```diff
@@ -142,9 +143,7 @@ private void resolveRoleNames(Set<String> roleNames, ActionListener<RolesRetriev
             Set<RoleDescriptor> roleDescriptors = rolesRetrievalResult.getRoleDescriptors();
             if (roleDescriptors.stream().anyMatch(RoleDescriptor::isUsingDocumentOrFieldLevelSecurity)
                 && DOCUMENT_LEVEL_SECURITY_FEATURE.checkWithoutTracking(licenseState) == false) {
-                effectiveDescriptors = roleDescriptors.stream()
-                    .filter(not(RoleDescriptor::isUsingDocumentOrFieldLevelSecurity))
-                    .collect(Collectors.toSet());
+                effectiveDescriptors = skipRolesUsingDocumentOrFieldLevelSecurity(roleDescriptors);
             } else {
                 effectiveDescriptors = roleDescriptors;
             }
```

Comment:
I'm good with the `maybeSkipRolesUsingDocumentOrFieldLevelSecurity` suggestion. My thinking around the `skip...` was that it's an internal, private method and as such can rely on the assumptions already established by the calling context. 

`maybeSkipRolesUsingDocumentOrFieldLevelSecurity` is indeed more self-contained though 👍

Reply:
From what I can read, having the warning header would not be very useful if it just happens for the first user query, but I don't have visibility on what the user experience would be in Kibana.

The other issue is that the warning is not actionable by the user, since it would require fixing a role or the license and they could be frustrated if they get it in the wrong place.

I'm fine with not having the warning header.